File: Source/Java/Expressive/src/main/java/com/bijington/expressive/operators/relational/LessThanOrEqualOperator.java
package com.bijington.expressive.operators.relational;

import com.bijington.expressive.expressions.BinaryExpression;
import com.bijington.expressive.expressions.BinaryExpressionType;
import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.operators.OperatorBase;
import com.bijington.expressive.operators.OperatorPrecedence;

/**
 * Created by shaun on 11/07/2016.
 */
public class LessThanOrEqualOperator extends OperatorBase {
    @Override
    public String[] getTags() { return new String[] { "<=" }; }

    @Override
    public IExpression buildExpression(String previousToken, IExpression[] expressions) {
        return new BinaryExpression(BinaryExpressionType.LessThanOrEqual, expressions[0], expressions[1]);
    }

    @Override
    public int getPrecedence(String previousToken) {
        return OperatorPrecedence.LessThanOrEqual;
    }
}

File: Source/CSharp/Expressive/Expressive/Operators/Relational/LessThanOrEqualOperator.cs
using Expressive.Expressions;

namespace Expressive.Operators.Relational
{
    internal class LessThanOrEqualOperator : OperatorBase
    {
        #region OperatorBase Members

        public override string[] Tags { get { return new[] { "<=" }; } }

        public override IExpression BuildExpression(Token previousToken, IExpression[] expressions, ExpressiveOptions options)
        {
            return new BinaryExpression(BinaryExpressionType.LessThanOrEqual, expressions[0], expressions[1], options);
        }

        public override OperatorPrecedence GetPrecedence(Token previousToken)
        {
            return OperatorPrecedence.LessThanOrEqual;
        }

        #endregion
    }
}
